src/main/resources/default/assets/common/core.js

```diff
@@ -71,3 +71,92 @@ sirius.postJSON = function(url, params) {
         return response.json();
     });
 }
+
+/**
+ * Throttles a javascript function by calling it only every x milliseconds.
+ * <p>
+ * The throttled function is called once immediately, and then every x milliseconds.
+ *
+ * @param fn        the function to be throttled
+ * @param threshold the time delaying each function call (in milliseconds).
+ */
+sirius.throttle = function (fn, threshold) {
```

Comment:
can we add a boolean to control if a leading call happens or not?

Reply:
In what use case do you not want to have an immediate reaction to an event?

Reply:
since these methods use cases are so universal, there should be a lot of use cases for any configuration, e.g. if you want to be "snappy" for the first interaction but still want to throttle in the long run, but we can add these options later when we neeed them